location/src/iosMain/kotlin/location/LocationManager.kt

```diff
@@ -0,0 +1,142 @@
+import com.splendo.kaluga.utils.byOrdinalOrDefault
+import platform.CoreLocation.CLAuthorizationStatus
+import platform.CoreLocation.CLLocation
+import platform.CoreLocation.CLLocationManager
+import platform.CoreLocation.CLLocationManagerDelegateProtocol
+import platform.Foundation.NSBundle
+import platform.Foundation.NSError
+import platform.darwin.NSObject
+
+actual class LocationManager(
+    private val locationManager: CLLocationManager,
+    locationPermission: Permission.Location,
+    permissions: Permissions,
+    autoRequestPermission: Boolean,
+    autoEnableLocations: Boolean,
+    locationStateRepo: LocationStateRepo
+) : BaseLocationManager(locationPermission, permissions, autoRequestPermission, autoEnableLocations, locationStateRepo) {
+
+    class Builder(private val locationManager: CLLocationManager = CLLocationManager()) : BaseLocationManager.Builder {
+
+        override fun create(
+            locationPermission: Permission.Location,
+            permissions: Permissions,
+            autoRequestPermission: Boolean,
+            autoEnableLocations: Boolean,
+            locationStateRepo: LocationStateRepo
+        ): BaseLocationManager = LocationManager(
+            locationManager,
+            locationPermission,
+            permissions,
+            autoRequestPermission,
+            autoEnableLocations,
+            locationStateRepo
+        )
+    }
+
+    private val locationManagerDelegate = object : NSObject(), CLLocationManagerDelegateProtocol {
+
+        override fun locationManager(manager: CLLocationManager, didUpdateLocations: List<*>) {
+            if (isMonitoringLocationUpdate) {
+                val locations = didUpdateLocations.mapNotNull { (it as? CLLocation)?.knownLocation }
+                if (locations.isNotEmpty()) {
+                    handleLocationChanged(locations)
+                }
+            }
+        }
+
+        override fun locationManager(manager: CLLocationManager, didUpdateToLocation: CLLocation, fromLocation: CLLocation) {
+            if (isMonitoringLocationUpdate) {
+                handleLocationChanged(listOf(didUpdateToLocation.knownLocation))
+            }
+        }
+
+        override fun locationManager(manager: CLLocationManager, didFinishDeferredUpdatesWithError: NSError?) {
+        }
+
+        override fun locationManager(manager: CLLocationManager, didChangeAuthorizationStatus: CLAuthorizationStatus) {
+            if (isMonitoringLocationEnable) {
+                handleLocationEnabledChanged()
+            }
+        }
+    }
+
+    private val authorizationStatus: CLAuthorizationStatusKotlin
+        get() = Enum.byOrdinalOrDefault(
+            CLLocationManager.authorizationStatus(),
+            CLAuthorizationStatusKotlin.notDetermined
+        )
+
+    var isMonitoringLocationEnable = false
+    var isMonitoringLocationUpdate = false
+
+    override suspend fun startMonitoringLocationEnabled() {
```

Comment:
What the difference of `startMonitoringLocation`?

Reply:
One is checking for whether GPS is enabled, the other is monitoring the actual location

Reply:
Don't get it, both functions are call `locationManager.startUpdatingLocation` to monitoring location

Reply:
Thats because the monitor enabled / monitor location call is the same on iOS. Its not on Android